tapdb/asset_minting.go

```diff
@@ -1470,6 +1470,70 @@ func (a *AssetMintingStore) CommitBatchTx(ctx context.Context,
 	})
 }
 
+// FetchDelegationKey fetches the delegation key for the given asset group
+// public key.
+func (a *AssetMintingStore) FetchDelegationKey(ctx context.Context,
+	groupKey btcec.PublicKey) (fn.Option[tapgarden.DelegationKey], error) {
+
+	var zero fn.Option[tapgarden.DelegationKey]
+	groupKeyBytes := groupKey.SerializeCompressed()
+
+	var delegationKey fn.Option[tapgarden.DelegationKey]
+
+	readOpts := NewAssetStoreReadTx()
+	dbErr := a.db.ExecTx(ctx, &readOpts, func(q PendingAssetStore) error {
+		fetchRow, err := q.FetchMintAnchorUniCommitment(
+			ctx, FetchPreCommitParams{
+				GroupKey: groupKeyBytes,
+			},
+		)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil
+			}
+			return fmt.Errorf("unable to fetch mint anchor "+
+				"uni commitment by group key: %w", err)
+		}
+
+		// If we didn't find any pre-commitment outputs, then
+		// we can return early.
+		if len(fetchRow) == 0 {
+			return nil
+		}
+
+		// If there are multiple pre-commitment outputs, just choose any
+		// single one, we assume they all have the same delegation key.
+		row := fetchRow[0]
+
+		pubKey, err := btcec.ParsePubKey(row.TaprootInternalKeyRaw)
+		if err != nil {
+			return fmt.Errorf("failed to parse taproot "+
+				"internal raw key: %w", err)
+		}
+
+		dKey := tapgarden.DelegationKey{
+			KeyLocator: keychain.KeyLocator{
+				Family: keychain.KeyFamily(
+					row.TaprootInternalKeyFamily.Int32,
+				),
+				Index: uint32(
+					row.TaprootInternalKeyIndex.Int32,
+				),
+			},
+			PubKey: pubKey,
+		}
+
+		delegationKey = fn.Some(dKey)
+
+		return nil
+	})
+	if dbErr != nil {
+		return zero, dbErr
```

Comment:
At this point `zero` and `delegationKey` should have the same, empty value. I think we can simplify to just use one.